crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java

```diff
@@ -80,27 +83,57 @@ private void appendHeading(Crd crd, String name) throws IOException {
     }
 
     private void usedIn(Class<?> cls, Map<Class<?>, Set<Class<?>>> usedIn) {
+        Set<Property> memorableProperties = new HashSet<>();
+
         for (Property property : properties(cls).values()) {
             if (property.isAnnotationPresent(KubeLink.class)) {
                 continue;
             }
+
+            OneOfType oneOfType = property.getAnnotation(OneOfType.class);
+            if (oneOfType != null && oneOfType.value().length > 0) {
+                for (OneOfType.Alternative alt : oneOfType.value()) {
+                    for (OneOfType.Alternative.Field field : alt.value()) {
+                        try {
+                            memorableProperties.add(new Property(property.getType().getType().getDeclaredField(field.value())));
+                        } catch (NoSuchFieldException e) {
+                            throw new RuntimeException("Failed to find field used in OneOfType annotation", e);
+                        }
+                    }
+                }
+            } else {
+                memorableProperties.add(property);
+            }
+        }
+
+        for (Property property : memorableProperties) {
             PropertyType propertyType = property.getType();
             Class<?> type = propertyType.isArray() ? propertyType.arrayBase() : propertyType.getType();
             for (Class<?> c : subtypesOrSelf(type)) {
                 if (Schema.isJsonScalarType(c)) {
                     continue;
                 }
-                Set<Class<?>> classes = usedIn.get(c);
-                if (classes == null) {
-                    classes = new HashSet<>(2);
-                    usedIn.put(c, classes);
-                }
+
+                Set<Class<?>> classes = getOrCreateClassesSet(c, usedIn);
+
                 classes.add(cls);
+
                 usedIn(c, usedIn);
             }
         }
     }
 
+    private Set<Class<?>> getOrCreateClassesSet(Class<?> c, Map<Class<?>, Set<Class<?>>> usedIn)   {
+        Set<Class<?>> classes = usedIn.get(c);
+
+        if (classes == null) {
+            classes = new HashSet<>(2);
```

Comment:
What's the reasoning behind initializing with this size?

Reply:
Honestly, it was in the old code. I just move it into separate method because of spotbugs. But I do not see any reason for it, so I changed it to 1.